engine/class_modules/sc_death_knight.cpp

```diff
@@ -4379,6 +4372,51 @@ struct defile_damage_t : public death_and_decay_damage_base_t
   }
 };
 
+struct deaths_due_damage_t : public death_and_decay_damage_base_t
```

Comment:
If it works with pestilence just like regular death and decay, you could make it a ``death_and_decay_damage_t`` instead of a ``death_and_decay_damage_base_t``

Reply:
Just re-tested to make sure, and yes, pestilence works with it.  Subclassing from death_and_decay_damage_t instead as per suggestion.

Reply:
```
Pre refactor
   deaths_due                  Count=   6.6| 34.786sec  DPE=3240| 2.15%  DPET=2399  DPR=   0  pDPS= 72  Miss= 0.00%  Hit=    58|    55|    61  Crit=   116|   109|   122|11.62%
    deaths_due                : start=  6.6 refresh= 82.2 interval= 35.1 trigger=  2.3 duration= 17.7 uptime= 39.31%

Post refactor
   deaths_due                  Count=   6.5| 34.915sec  DPE=3246| 2.15%  DPET=2404  DPR=   0  pDPS= 72  Miss= 0.00%  Hit=    58|    55|    61  Crit=   116|   109|   122|11.68%
    deaths_due                : start=  6.5 refresh= 82.4 interval= 35.1 trigger=  2.3 duration= 17.7 uptime= 39.24%
```